### ISSUE TITLE
Updating to Android 6.0.0-beta.5 telemetery service constantly crashes

### ISSUE THREAD

**Issue by sigis151** [mapbox/mapbox-gl-native#11554]
After upgrading to Android 6.0.0-beta.5 telemetry service constantly crashes with two different exceptions

the crash happens on several devices: P9 lite, Nexus 5x, S8. with android 6.0 and up

I could not stably reproduce it yet. Second exception might be related to leak canary running on different process.
```
    Process: ***, PID: 5453
    java.lang.RuntimeException: Unable to start service com.mapbox.android.telemetry.TelemetryService@1091233 with null: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.content.Intent.getBooleanExtra(java.lang.String, boolean)' on a null object reference
        at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3491)
        at android.app.ActivityThread.-wrap20(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1692)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.content.Intent.getBooleanExtra(java.lang.String, boolean)' on a null object reference
        at com.mapbox.android.telemetry.TelemetryService.enableTelemetryLocationState(TelemetryService.java:203)
        at com.mapbox.android.telemetry.TelemetryService.onStartCommand(TelemetryService.java:48)
        at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3474)
        at android.app.ActivityThread.-wrap20(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1692) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6494) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
		
```
```	
    java.lang.ClassCastException: android.os.BinderProxy cannot be cast to com.mapbox.android.telemetry.TelemetryService$TelemetryBinder
        at com.mapbox.android.telemetry.MapboxTelemetry$3.onServiceConnected(MapboxTelemetry.java:375)
        at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:1652)
        at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1681)
        at android.os.Handler.handleCallback(Handler.java:790)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

**Comment by LukasPaczos** [comment]
Thanks for reporting this issue @sigis151!
Following up in https://github.com/mapbox/mapbox-events-android/issues/97.

/cc @Guardiola31337 @electrostat

**Comment by Guardiola31337** [comment]
https://github.com/mapbox/mapbox-events-android/pull/98 landed and it'll be included in the next beta release.
We'll comment here when that happens so we'll be able to include that fixed version in the Maps SDK and close OP after releasing a new version of the Maps SDK including the fixed dependency.

cc @LukasPaczos @electrostat

**Comment by LukasPaczos** [comment]
Closing as the fix landed with `6.0.0-beta.6`.
